### PR TITLE
Can edit puzzles

### PR DESCRIPTION
**Pull request by paymog**
Can edit everything property of a cryptogram except for its user or its
id

### CODE REVIEW

File: src/tests/persistence/StubDB.java [hunk 1]
```diff
@@ -43,6 +43,83 @@ public boolean savePuzzle(Puzzle puzzle)
...
+	}
+
+	@Override
+	public boolean updatePlaintext(long id, String newPlaintext)
+	{
+		Puzzle puz = this.findPuzzle(id);
+		if (puz == null)
+		{
+			return false;
+		}
```
Comment (paymog) [inline src/tests/persistence/StubDB.java:116]: At this point you've probably noticed the same 4 lines being written over and over. Can you think of a better way to write this code to reduce the duplication?

Comment (rikimbo) [inline src/tests/persistence/StubDB.java:116]: I can't think of anything that would easily reduce those lines, or factor them out.  I don't think something like this is that big of a deal, though.

**Comment by willvoutier** [comment]
@paymog 

I would prefer a different (invisible) interface for updating. Since we're going to be re-using the creation screen, it'd be nice to not have to change it too much.

Do you think we could just do some `set[Attribute]()`'s on a puzzle that is already created, and then call `save()`?

The save would just need to figure out if the puzzle is new or updated (whether it has an `ID`).

Thoughts?

**Comment by rikimbo** [comment]
I agree with what Will said.  That's how I'd think of doing it too, if I was implementing the edit feature.

**Comment by paymog** [comment]
So what would the flow of logic be?

We select a puzzle
Press the edit button
Change stuff
Press save button -> this creates a new puzzle?
Save the newly created puzzle
At this point, the database (or PuzzleManager) should be able to detect a modified puzzle vs a new puzzle and act accordingly?

**Comment by willvoutier** [comment]
Not quite. 

We select a puzzle
Press the edit button
Change stuff
Press save button -> the save will check to see if the puzzle has an ID or not (if it has one already, it'll update, otherwise it'll insert)

**Comment by willvoutier** [comment]
Looks good.

**Pull request merged**